packages/amplify-graphql-auth-transformer/src/resolvers/subscriptions.ts

```diff
@@ -120,9 +128,27 @@ const combineAuthExpressionAndFilter = (ownerExpression: Array<Expression>, grou
     raw('$authGroupRuntimeFilter.size() > 0'),
     qref(methodCall(ref('authRuntimeFilter.addAll'), ref('authGroupRuntimeFilter'))),
   ),
+  // isOwnerAuthAuthorizedAndNoOtherFilter is defined as user authorized
+  // with an owner param which we've verified matches their token claims,
+  // and they have aonly a single rtf filter (e.g. their user) and no
+  // additional input filters. In this case we can rely on
+  // Subscription basic filtering which is computationally less expensive
+  // than Runtime filtering, and faster for the customer.
+  set(
+    ref(IS_OWNER_AUTH_AUTHORIZED_AND_NO_OTHER_FILTERS_FLAG),
+    and([
+      ref(HAS_VALID_OWNER_ARGUMENT_FLAG),
+      equals(methodCall(ref('authRuntimeFilter.size')), int(1)),
+      equals(methodCall(ref('ctx.args.filter.size')), int(0)),
```

Comment:
could this cause nullreference exception if no `filter` exists in `ctx.args`?

Reply:
Hmm, I haven't witnessed that in testing, let me confirm w/ logs

Reply:
VTL could swallow these errors and not log anywhere. I also recommend to test this with both variables and inline argument syntax.

Reply:
Yep, wrapping in a null check up front now (yay, state mutation!)